src/analysis/retail/mage/frost/core/IcyVeins.tsx

```diff
@@ -28,17 +35,25 @@ class IcyVeins extends Analyzer {
   activeTime: number[] = [];
   buffApplies: number = 0;
 
+  castEntries: BoxRowEntry[] = [];
+
   constructor(options: Options) {
     super(options);
     this.addEventListener(
-      Events.applybuff.by(SELECTED_PLAYER).spell(TALENTS.ICY_VEINS_TALENT),
+      Events.applybuff.by(SELECTED_PLAYER).spell(TALENTS_MAGE.ICY_VEINS_TALENT),
       this.onIcyVeinsStart,
     );
     this.addEventListener(
-      Events.removebuff.by(SELECTED_PLAYER).spell(TALENTS.ICY_VEINS_TALENT),
+      Events.removebuff.by(SELECTED_PLAYER).spell(TALENTS_MAGE.ICY_VEINS_TALENT),
       this.onIcyVeinsEnd,
     );
     this.addEventListener(Events.fightend, this.onFightEnd);
+    // Guide
+    this.addEventListener(
+      Events.removebuff.by(SELECTED_PLAYER).spell(TALENTS_MAGE.ICY_VEINS_TALENT),
+      this.generateCastEntryOnBuffEnd,
+    );
+    this.addEventListener(Events.fightend, this.generateCastEntryOnFightEnd);
```

Comment:
Pretty sure this will also pickup Time Anomaly procs, which may not be desired.

My first thought on how to handle this is to add a `SPELL_CAST` link to `CastLinkNormalizer` for Icy Veins, allowing you to differentiate casts vs procs on the applybuff, and then you can check that link and disregard procs. A possible issue with that could be a bad linkage if player gets TA proc and then activates Icy Veins hardcast (I simply don't know how that will look in logs and it might mess with things)

Reply:
Thanks! I will look on what you said. I think I have some TA logs to test this situation. I'll let you know what I found....

Reply:
Do as you said, and test it on raid. Waited until proc pops and after cast IV. It worked.
Buff (TA proc) started at 0:05, and I casted at 0:08. The analysis only watch for the "casted" portion.
![image](https://github.com/WoWAnalyzer/WoWAnalyzer/assets/32552126/e1c446ca-2847-474b-9e96-8f4a834703ee)
![image](https://github.com/WoWAnalyzer/WoWAnalyzer/assets/32552126/10bc1bbe-5711-482a-ab9e-69b3c564ee34)